### ISSUE TITLE
Incremental compilation ignores changes in constructor parameters

### ISSUE THREAD

**Issue by dos65** [lampepfl/dotty#12898]
## Minimized code

First, compile the following sources:
```scala
// src/main/scala/Foo.scala
class Foo(a: Int, b: String) {
  def foo: String = ???
}

// src/main/scala/Main.scala
class Main {
  val foo = Foo(1, "asd").foo
}
// compile: OK
```

Then add an additional parameter to `Foo` class and compile one more time:
```scala
// src/main/scala/Foo.scala
class Foo(a: Int, b: String, c: Boolean) {
  def foo: String = ???
}

// src/main/scala/Main.scala
class Main {
  val foo = Foo(1, "asd").foo // <<- no errors here
}
// compile: OK
```

## Expectation

The second compilation run should fail

**Comment by bishabosha** [comment]
this was fixed in #12712

**Comment by smarter** [comment]
Let's close then?
